tests/v2/validation/fleet/provisioning/airgap/fleet_in_airgap_test.go

```diff
@@ -0,0 +1,214 @@
+	require.NoError(a.T(), err)
+
+	a.corralPackage = corral.PackagesConfig()
+
+	bastionIP := corralConfig.CorralConfigVars[corralBastionIP]
+	bastionUser := corralConfig.CorralConfigVars[corralSSHUser]
+
+	// format the privateKey string for use with ssh package.
+	privateKey := corralConfig.CorralConfigVars[corralPrivateKey]
+	privateKey = strings.Replace(privateKey, "\\n", "\n", -1)
+	privateKey = strings.Replace(privateKey, "\"", "", -1)
+	privateKey = fmt.Sprintf("%s", privateKey)
+
+	err = corral.UpdateCorralConfig(corralBastionIP, bastionIP)
+	require.NoError(a.T(), err)
+
+	registrySetting, err := client.Management.Setting.ByID(systemRegistry)
+	require.NoError(a.T(), err)
+
+	a.registryFQDN = registrySetting.Value
+
+	internalIP, err := setupAirgapFleetResources(bastionUser, bastionIP, privateKey)
+	require.NoError(a.T(), err, "failed to setup local git repo on bastion")
+
+	fleetSecretName, err := createFleetSSHSecret(client, privateKey)
+	require.NoError(a.T(), err, "failed to create SSH Secrets for fleet")
+
+	a.fleetGitRepo = &v1alpha1.GitRepo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fleet.FleetMetaName + namegenerator.RandStringLower(5),
+			Namespace: fleet.Namespace,
+		},
+		Spec: v1alpha1.GitRepoSpec{
+			Repo:            fmt.Sprintf("%s@%s:/home/%s/%s", bastionUser, internalIP, bastionUser, fleetExampleFolderName),
+			Branch:          fleet.BranchName,
+			Paths:           []string{fleet.GitRepoPathLinux},
+			CorrectDrift:    &v1alpha1.CorrectDrift{},
+			ImageScanCommit: v1alpha1.CommitSpec{AuthorName: "", AuthorEmail: ""},
+			Targets: []v1alpha1.GitTarget{
+				{
+					ClusterSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      fleet.MatchKey,
+								Operator: fleet.MatchOperator,
+								Values: []string{
+									fleet.HarvesterName,
+								},
+							},
+						},
+					},
+				},
+			},
+			ClientSecretName: fleetSecretName,
+		},
+	}
+
+	a.client, err = a.client.ReLogin()
+	require.NoError(a.T(), err)
+
+	if a.clustersConfig.RKE2KubernetesVersions == nil {
+		rke2Versions, err := kubernetesversions.ListRKE2AllVersions(a.client)
+		require.NoError(a.T(), err)
+
+		a.clustersConfig.RKE2KubernetesVersions = []string{rke2Versions[len(rke2Versions)-1]}
+	}
+
+	if a.clustersConfig.CNIs == nil {
+		a.clustersConfig.CNIs = []string{fleet.CniCalico}
+	}
+}
+
+func (a *AirGapRKE2CustomClusterTestSuite) TestCustomClusterWithGitRepo() {
+	fleetVersion, err := fleet.GetDeploymentVersion(a.client, fleet.FleetControllerName, fleet.LocalName)
+	require.NoError(a.T(), err)
+	singleNodeRoles := []provisioninginput.MachinePools{provisioninginput.AllRolesMachinePool}
+
+	tests := []struct {
+		name        string
+		client      *rancher.Client
+		machinePool []provisioninginput.MachinePools
+	}{
+		{fleet.FleetName + " " + fleetVersion + "-" + permutations.RKE2AirgapCluster, a.standardClient, singleNodeRoles},
```

Comment:
as its only 1 test, why do we want a tdd test here?

Reply:
this was a discussion sometime last week. There are pros and cons to writing tests structured as a table test. I can ask in the channel, but my preference is to still write it as a table test as it allows for modularity in the future i.e. easily add a test to the table later, and it keeps a good format / practice going so that tests are written in a consistent manner.